lib/web/terminal.go

```diff
@@ -727,12 +720,13 @@ func (w *terminalStream) Read(out []byte) (n int, err error) {
 
 // Close the websocket.
 func (w *terminalStream) Close() error {
-	return w.ws.Close()
+	return w.ws.Close(websocket.StatusNormalClosure, "")
 }
 
 // deadlineForInterval returns a suitable network read deadline for a given ping interval.
 // We chose to take the current time plus twice the interval to allow the timeframe of one interval
 // to wait for a returned pong message.
+// TODO(zmb3): remove this??
```

Comment:
Still investigating this, but I think we'll be able to remove.

With this library, the `Ping()` method automatically waits for a pong, and if the pong doesn't respond in time the broken connection is detected, so we no longer need to do the hacky "extend the read deadline" every few seconds trick.